src/main/java/duke/data/storage/StorageManager.java

```diff
@@ -0,0 +1,155 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class StorageManager {
+
+    public static final String PARAM_DELIMIT_SAVE = " | ";
+
+    /* Symbol */
+    public static final String SYMBOL_TODO = "T";
+    public static final String SYMBOL_DEADLINE = "D";
+    public static final String SYMBOL_EVENT = "E";
+
+    /* Files */
+    private static final String RELATIVE_DIR = System.getProperty("user.dir");
+    private static final String FILE_SEPARATOR = File.separator;
+
+    private File dataFile;
+
+    /***
+     * StorageManager Constructor
+     * Check if directory folder exist else create folder
+     * Check if data file exist else create file
+     * @param fileName
+     */
+    public StorageManager(String fileName){
+        this.dataFile = new File(prepareFile(fileName));
+        try{
+            if (!dataFile.getParentFile().exists()) {
+                dataFile.getParentFile().mkdirs();
+            }
+
+            if (!dataFile.exists()) {
+                dataFile.createNewFile();
+            }
+        }catch (IOException e){
+            System.out.println("Cannot create file; reason: " + e.getMessage());
+        }
+
+    }
+
+    /***
+     * Change String into file format
+     * @param fileName
+     * @return
+     */
+    private String prepareFile(String fileName){
+        return RELATIVE_DIR + fileName.replace("\\",FILE_SEPARATOR).replace("/",FILE_SEPARATOR);
+    }
+
+    /***
+     * Read file and store into string arraylist
+     * Parse the stringCommand into task
+     * Keep the task into tasklist and return it
+     * @return Tasklist to be loaded on startup
+     * @throws IOException : files error
+     */
+    public ArrayList<Task> load() throws IOException{
+        ArrayList<Task> taskList = null;
+        try {
+            ArrayList<String> dataItems = readFile();
+            taskList = parse(dataItems);
+        } catch (IOException e) {
+            throw e;
+        }
+        return taskList;
+    }
+
+    /***
+     * Read file into string arraylist
+     * @return Arraylist String data item
+     * @throws IOException : throws file empty or file don't exist
+     */
+    private ArrayList readFile() throws IOException {
+        if (!dataFile.exists()) {
+            throw new FileNotFoundException();
+        }
+        if (dataFile.length() == 0) {
+            throw new IOException();
+        }
+        ArrayList<String> dataItems = (ArrayList) Files.readAllLines(dataFile.toPath(), Charset.defaultCharset());
+
+        return dataItems;
+    }
+
+    /***
+     * Save content provided into save file
+     * @param saveContent : Formated task data item
+     */
+    public void save(String saveContent) {
+//        System.out.println("\t Saving to :\""+ fileName +"\""+ Messages.LS);
```

Comment:
You may want to remove old code instead of commenting them out before making the PR